script/validate-electron-version.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable no-sync */
+/// <reference path="./globals.d.ts" />
+
+import * as distInfo from './dist-info'
+
+type ChannelToValidate = 'production' | 'beta'
+
+/**
+ * This object states the valid/expected Electron versions for each publishable
+ * channel of GitHub Desktop.
+ *
+ * The purpose of this is to ensure that we don't accidentally publish a
+ * production/beta/test build with the wrong version of Electron, which could
+ * cause really bad regressions to our users, and also the inability to go back
+ * to a previous version of GitHub Desktop without losing all settings.
+ */
+const ValidElectronVersions: Record<ChannelToValidate, string> = {
+  production: '19.0.0',
+  beta: '19.0.0',
+}
+
+const channel = getChannelToValidate()
+
+if (channel === null) {
+  console.log(
+    `No need to validate the Electron version of a ${distInfo.getChannel()} build.`
+  )
+  process.exit(0)
+}
+
+const expectedVersion = ValidElectronVersions[channel]
+const pkg: Package = require('../package.json')
+const actualVersion = pkg.devDependencies?.electron
```

Comment:
Minor concern - If someone uses the  `=19.0.0` in the package.json as opposed to the `19.0.0` will this fail?  We could just update one or the other if that were to occur, but, we could use the `semver` package to do the comparison like in the `endpoint-capabilities` file to not hit that bump/have to remember which one we used.

Reply:
Using `semver` sounds like a nice idea! I kind of wanted to enforce the exact version to use, so didn't even consider other options.

Reply:
After looking into this, I think it makes sense to be explicit about what we want. According to [the documentation](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#dependencies), we shouldn't use `=version` when we want to require a specific version, and I'm a bit concerned about letting `semver` doing something "smarter" by mistake (like considering the same major version but accepting patch revisions, for example).

So I think comparing these hardcoded strings should be more than fine. Do you agree? Or do you feel strongly that we should rely on `semver` for this?

Reply:
>So I think comparing these hardcoded strings should be more than fine. Do you agree?

Yes. They are more than fine. :) It is just a convenience factor.

Reason I brought it up is because I have seen in our history some devs have used `=` and some not. The documentation you referenced says to refer to [this semver documentation](https://github.com/npm/node-semver#ranges) for more on semver notation. This says using the `=` is valid and I think a sort of the "default" syntax as the doc says to assume it is present if not provided. The semver library has a `semver.eq()` which would make sure that for a version number it is its equivalent (as opposed to `satifies()`. I believe `v19.0.0`, `=19.0.0`, and `19.0.0` are all equal in terms of versioning.  (Tho I haven't seen v19.0.0 used before)

For reference:
![image](https://user-images.githubusercontent.com/75402236/213206518-d3233682-0582-4dc5-acff-cbcecbca5db5.png)

Reply:
As discussed privately, `eq` doesn't seem to work well with `=...` versions :(

![image](https://user-images.githubusercontent.com/1083228/213248299-945442e3-a3f7-4766-b7b5-c108be01d84c.png)

So we decided to use `===` which is completely clear and unambiguous, and with that avoid any possible confusion.